diff --git a/src/hnsw/hnsw_optimize_join.cpp b/src/hnsw/hnsw_optimize_join.cpp
index 36c220f..21a907f 100644
--- a/src/hnsw/hnsw_optimize_join.cpp
+++ b/src/hnsw/hnsw_optimize_join.cpp
@@ -449,10 +449,10 @@ bool HNSWIndexJoinOptimizer::TryOptimize(Binder &binder, ClientContext &context,
 		return false;
 	}
 	const auto &compare_expr = filter.expressions.back()->Cast<BoundComparisonExpression>();
-	if (compare_expr.right->GetExpressionType() != ExpressionType::VALUE_CONSTANT) {
+	if (compare_expr.Right().GetExpressionType() != ExpressionType::VALUE_CONSTANT) {
 		return false;
 	}
-	const auto &constant_expr = compare_expr.right->Cast<BoundConstantExpression>();
+	const auto &constant_expr = compare_expr.Right().Cast<BoundConstantExpression>();
 	if (constant_expr.GetReturnType() != LogicalType::BIGINT) {
 		return false;
 	}
@@ -461,10 +461,10 @@ bool HNSWIndexJoinOptimizer::TryOptimize(Binder &binder, ClientContext &context,
 		// Can only optimize up to SVS
 		return false;
 	}
-	if (compare_expr.left->GetExpressionType() != ExpressionType::BOUND_COLUMN_REF) {
+	if (compare_expr.Left().GetExpressionType() != ExpressionType::BOUND_COLUMN_REF) {
 		return false;
 	}
-	auto &filter_ref_expr = compare_expr.left->Cast<BoundColumnRefExpression>();
+	auto &filter_ref_expr = compare_expr.Left().Cast<BoundColumnRefExpression>();
 	if (filter_ref_expr.binding.table_index != window.window_index) {
 		return false;
 	}
